### PR TITLE
SPLASH-33: Library UI & Logic

### PR DESCRIPTION
**Pull request by Pepegaga**

### CODE REVIEW

File: src/app/library/components/add-playlist.component.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import styled from 'styled-components/native';
+
+import I18n from '../../utils/i18n';
+
+import { PlaylistContainer, PlaylistImage, InfoWrapper, PlaylistName } from './playlist-item.component';
```
Comment (TheLonelyAstronaut) [inline src/app/library/components/add-playlist.component.tsx]: If you using same styled components, create file and import it from playlist-item.component and add-playlist.componetn

File: src/app/library/components/add-playlist.component.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+export const AddPlaylistName = styled(PlaylistName)`
+    margin-top: ${(props) => props.theme.spacer * 1.5};
+`;
+
+export const AddPlaylistItem: React.FC<Props> = (props: Props) => {
+    return (
+        <PlaylistContainer onPress={props.onPress}>
+            <PlaylistImage source={require('../../../assets/add-icon.png')} />
```
Comment (TheLonelyAstronaut) [inline src/app/library/components/add-playlist.component.tsx]: Here you should create styled component with + icon from vector icons package

File: src/app/library/components/playlist-item.component.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+export const PlaylistContainer = styled.TouchableOpacity`
+    width: ${DEVICE_SIZE.width};
+    height: ${DEVICE_SIZE.height * 0.09};
+    background-color: ${(props) => props.theme.colors.screenBackground};
+    margin-bottom: ${(props) => props.theme.spacer * 3}px;
+    flex-direction: row;
+`;
+
+export const PlaylistImage = styled.Image`
```
Comment (TheLonelyAstronaut) [inline src/app/library/components/playlist-item.component.tsx]: Image from /ui

File: src/app/library/components/playlist-item.component.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    color: ${(props) => props.theme.colors.secondary};
+    font-size: ${(props) => props.theme.fontSize.small};
+`;
+export const CombinedImageContainer = styled.View`
+    width: 90px;
+    height: 90px;
+    background-color: white;
+    margin-left: ${(props) => props.theme.spacer * 3}px;
+`;
+export const Img = styled.Image`
```
Comment (TheLonelyAstronaut) [inline src/app/library/components/playlist-item.component.tsx]: Looks ugly, rename

File: src/app/library/sagas.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { SagaIterator } from 'redux-saga';
+import { call, put, takeLatest } from 'redux-saga/effects';
+
+import { client } from '../../graphql/api';
+
+import { ADD_PLAYLIST, LOAD_LIBRARY } from './actions';
+
+export function* loadLibrarySaga(action: ReturnType<typeof LOAD_LIBRARY.TRIGGER>): SagaIterator {
+    try {
+        const result = yield call(client.getLibrary, action.payload);
```
Comment (TheLonelyAstronaut) [inline src/app/library/sagas.ts:12]: You should add .STARTED, when you setting isFetching to true, also you should display ActivityIndicator in the screen

File: src/app/library/sagas.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        yield put(LOAD_LIBRARY.COMPLETED(result));
+    } catch (e) {
+        console.log(e);
+    }
+}
+
+export function* addPlaylistSaga(action: ReturnType<typeof ADD_PLAYLIST.TRIGGER>): SagaIterator {
+    try {
+        const result = yield call(client.addPlaylist, action.payload);
```
Comment (TheLonelyAstronaut) [inline src/app/library/sagas.ts:23]: Same

File: src/app/library/sagas.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+import { client } from '../../graphql/api';
+
+import { ADD_PLAYLIST, LOAD_LIBRARY } from './actions';
+
+export function* loadLibrarySaga(action: ReturnType<typeof LOAD_LIBRARY.TRIGGER>): SagaIterator {
+    try {
+        const result = yield call(client.getLibrary, action.payload);
+        yield put(LOAD_LIBRARY.COMPLETED(result));
+    } catch (e) {
+        console.log(e);
```
Comment (TheLonelyAstronaut) [inline src/app/library/sagas.ts]: Also you should trigger .COMPLETED.failed() here

File: src/app/library/sagas.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    }
+}
+
+export function* addPlaylistSaga(action: ReturnType<typeof ADD_PLAYLIST.TRIGGER>): SagaIterator {
+    try {
+        const result = yield call(client.addPlaylist, action.payload);
+        yield put(LOAD_LIBRARY.COMPLETED(result));
+    } catch (e) {
+        console.log(e);
```
Comment (TheLonelyAstronaut) [inline src/app/library/sagas.ts]: Same

File: src/app/library/screens/library-screen.component.tsx [hunk 9]
```diff
@@ -1,12 +1,122 @@
...
 export const LibraryScreen: React.FC = () => {
+    const [visible, setVisible] = useState(false);
+    const [playlistName, setPlaylistName] = useState('');
+    const dispatch = useDispatch();
+    const theme = useTheme();
+
+    const handleAddPlaylist = useCallback(
+        (name: string) => {
+            dispatch(
+                ADD_PLAYLIST.TRIGGER({ type: LibraryElementType.PLAYLIST, data: { name: name, id: 15, tracks: [] } })
```
Comment (TheLonelyAstronaut) [inline src/app/library/screens/library-screen.component.tsx]: We should add playlist in the server, we can't generate ID on client. Add id on server side (GraphQLAPI now)

Comment (TheLonelyAstronaut) [reply src/app/library/screens/library-screen.component.tsx]: And remove tracks from here ofc, we can create playlist only in empty state, why we passing empty array here?)

File: src/app/library/screens/library-screen.component.tsx [hunk 10]
```diff
@@ -1,12 +1,122 @@
...
+            );
+        },
+        [dispatch]
+    );
+
+    useEffect(() => {
+        dispatch(LOAD_LIBRARY.COMPLETED(library));
+    }, [dispatch, handleAddPlaylist]);
```
Comment (TheLonelyAstronaut) [inline src/app/library/screens/library-screen.component.tsx]: Why we need handleAddPlaylist as dependency here?

File: src/app/library/sagas.ts (hunk 5 above)
Comment (Pepegaga) [reply src/app/library/sagas.ts:10]: Made it in TRIGGER action

**Pull request merged**